virtual_machine/main.tf

```diff
@@ -0,0 +1,57 @@
+resource "azurerm_public_ip" "virtual_machine" {
+  allocation_method   = "Static"
+  location            = var.location
+  name                = "${var.name}-public-ip"
+  resource_group_name = var.resource_group_name
+}
+
+resource "azurerm_network_interface" "virtual_machine" {
+  name                = "${var.name}-nic"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  ip_configuration {
+    name                          = "${var.name}-ip"
+    subnet_id                     = var.subnet_id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = azurerm_public_ip.virtual_machine.id
+  }
+}
+
+resource "azurerm_virtual_machine" "virtual_machine" {
```

Comment:
this resource is deprecated, add an option to create either a linux or windows vm using a non depricated resource